CMakeLists.txt

```diff
@@ -1,20 +1,28 @@
 cmake_minimum_required(VERSION 3.15...3.26)
 project(${SKBUILD_PROJECT_NAME} LANGUAGES NONE)
 
+find_package(
+  Python
+  COMPONENTS Interpreter
+  REQUIRED)
 
-set(idc_index_release_version "0.3.2")
-set(idc_index_data_url "https://github.com/ImagingDataCommons/idc-index/releases/download/${idc_index_release_version}/idc_index.csv.zip")
-set(idc_index_data_sha256 "70ec9f915686a27bee3098163b8695c69c8696c05bfb7bd76943a24024cdeeb9")
+if(NOT DEFINED ENV{GCP_PROJECT})
+  message(FATAL_ERROR "GCP_PROJECT env. variable is not set")
+endif()
 
-#
-# Download and install index
-#
 set(download_dir "${PROJECT_BINARY_DIR}")
-include(FetchContent)
-FetchContent_Populate(s5cmd
-  URL ${idc_index_data_url}
-  URL_HASH SHA256=${idc_index_data_sha256}
-  DOWNLOAD_DIR ${download_dir}
-  DOWNLOAD_NO_EXTRACT TRUE
-  )
-install(FILES "${download_dir}/idc_index.csv.zip" DESTINATION "idc_index_data")
+
+add_custom_command(
+   OUTPUT
+     ${download_dir}/idc_index.csv.zip
+     #${download_dir}/idc_index.parquet
```

Comment:
I was having a hard time passing the variable but I'll try again. I think going forward we may only have either csv or parquet but not both..@fedorov will that be true?

Reply:
As we need to parameterize on multiple files, it would be easier if we simply have either one of them and not both.

Reply:
> I think going forward we may only have either csv or parquet but not both..@fedorov will that be true?

Correct, this is my current thinking. I think we should go with CSV to minimize disruptions for the initial release, and in the following - still for IDC v17 - switch to Parquet. Those files are large, and we will have more queries/files in the future, so support of both is going to be prohibitive, and I do not think it would add value.

Reply:
Great, then I think other than the big picture about version handling and not uploading the tar.gz to PyPi, all other concerns may be resolved.

Reply:
> re: parquet

Instead of having commented code, I added the code I originally suggest in https://github.com/ImagingDataCommons/idc-index-data/pull/5#issuecomment-2000130263. we now have the option `IDC_INDEX_DATA_GENERATE_PARQUET` disabled by default.

Reply:
Thanks, JC! This is assuming we will always need csv right? And I think we may only need parquet in the future. I also just realized that the GHA invoked by this PR is not actually checking out the PR repo as you can see here. It is downloading the index from hardcoded URL. 

![image](https://github.com/ImagingDataCommons/idc-index-data/assets/115020590/dec71baf-f265-4436-becb-0c11c3ae3c37)

Reply:
> I think we may only need parquet in the future.

Also added option `IDC_INDEX_DATA_GENERATE_CSV_ARCHIVE` enabled by default

Reply:
> re: GHA invoked by this PR is not actually checking out the PR repo

Can you share a link ?

Reply:
every single invocation, so the checks on this repo may not be correct

https://github.com/ImagingDataCommons/idc-index-data/actions/runs/8397545642/job/23001014382

Reply:
This is a regressing introduced by the recent pull request switching from `pull_request` to `pull_request_target`, see https://stackoverflow.com/questions/74957218/what-is-the-difference-between-pull-request-and-pull-request-target-event-in-git